### PR TITLE
Subaru: move carspecs to PlatformConfig

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: selfdrive/car/__init__.py [hunk 1]
```diff
@@ -245,12 +245,21 @@ def update(self, current_value, current_counter):
 
 CarInfos = Union[CarInfo, List[CarInfo]]
 
+
+@dataclass
+class CarSpecs:
```
Comment (adeebshihadeh) [inline selfdrive/car/__init__.py:250]: any reason to make this its own struct?

File: selfdrive/car/subaru/values.py [hunk 2]
```diff
@@ -124,55 +128,66 @@ class CAR(Platforms):
...
   )
   FORESTER_HYBRID = SubaruPlatformConfig(
     "SUBARU FORESTER HYBRID 2020",
     SubaruCarInfo("Subaru Forester Hybrid 2020"),
     dbc_dict('subaru_global_2020_hybrid_generated', None),
-  )
-  FORESTER = SubaruPlatformConfig(
-    "SUBARU FORESTER 2019",
-    SubaruCarInfo("Subaru Forester 2019-21", "All"),
+    specs=FORESTER.specs,
```
Comment (jnewb1) [inline selfdrive/car/subaru/values.py:149]: it's easier to copy from another one

Comment (jnewb1) [reply selfdrive/car/subaru/values.py:149]: @adeebshihadeh

File: selfdrive/car/__init__.py (hunk 1 above)
Comment (jnewb1) [reply selfdrive/car/__init__.py:250]: https://github.com/commaai/openpilot/pull/31550#discussion_r1500130053

**Comment by jnewb1** [comment]
before it was done like
```
    elif candidate in (CAR.FORESTER, CAR.FORESTER_2022, CAR.FORESTER_HYBRID):
      ret.mass = 1568.
      ret.wheelbase = 2.67
      ret.centerToFront = ret.wheelbase * 0.5
      ret.steerRatio = 17           # learned, 14 stock
```

File: selfdrive/car/subaru/interface.py [hunk 3]
```diff
@@ -40,43 +40,32 @@ def _get_params(ret, candidate, fingerprint, car_fw, experimental_long, docs):
     else:
       CarInterfaceBase.configure_torque_tune(candidate, ret.lateralTuning)
 
+    ret.mass = candidate.config.specs.mass
```
Comment (adeebshihadeh) [inline selfdrive/car/subaru/interface.py:43]: move here? https://github.com/commaai/openpilot/blob/015eed5d61c583328d419a7990ffac0eee97edd4/selfdrive/car/interfaces.py#L110

Comment (jnewb1) [reply selfdrive/car/subaru/interface.py]: once all brands are done?

File: selfdrive/car/__init__.py (hunk 1 above)
Comment (adeebshihadeh) [reply selfdrive/car/__init__.py:250]: hmm, it's not super obvious that's happening. might be better to define them as constants above

File: selfdrive/car/subaru/interface.py (hunk 3 above)
Comment (adeebshihadeh) [reply selfdrive/car/subaru/interface.py]: why not now? just check if it’s none?

File: selfdrive/car/__init__.py (hunk 1 above)
Comment (sshane) [reply selfdrive/car/__init__.py:250]: I agree, it should be easy to group multiple platforms together with the same specs, that's one thing the old method made easy, so it should be as obvious and easy to do the same with the refactor. I'm not sure what that looks like yet though

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**